modules/mudletclient.go

```diff
@@ -0,0 +1,181 @@
+package modules
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/GoMudEngine/GoMud/internal/events"
+	"github.com/GoMudEngine/GoMud/internal/mudlog"
+	"github.com/GoMudEngine/GoMud/internal/plugins"
+	"github.com/GoMudEngine/GoMud/internal/users"
+)
+
+var (
+	//go:embed mudletclient/*
+	mudletclient_Files embed.FS
+	
+	// Instance accessible from elsewhere in this package
+	mudletClientModule MudletClientModule
+)
+
+func init() {
+	// Create the module instance first
+	mudletClientModule = MudletClientModule{
+		plug: plugins.New(`mudletclient`, `1.0`),
+	}
+
+	// Attach the file system to the plugin
+	if err := mudletClientModule.plug.AttachFileSystem(mudletclient_Files); err != nil {
+		panic(err)
+	}
+
+	// Set default values directly during initialization
+	mudletClientModule.setDefaultValues()
+
+	// Register the onLoad callback to load config values after plugin system is ready
+	mudletClientModule.plug.Callbacks.SetOnLoad(func() {
+		mudletClientModule.RefreshConfig()
+	})
+
+	// Export a function to register client info for Mudlet clients
+	mudletClientModule.plug.ExportFunction("SendClientInfo", mudletClientModule.SendClientInfoExported)
+}
+
+type MudletClientModule struct {
+	plug          *plugins.Plugin
+	uiUrl         string
+	uiVersion     string
+	mapUrl        string
+	mapperUrl     string
+	mapperVersion string
+	// Track which users already have data sent
+	registeredUsers map[int]bool
+}
+
+// Set hardcoded default values for initialization
+func (g *MudletClientModule) setDefaultValues() {
+	g.uiUrl = ""
+	g.uiVersion = "1.0.0"
+	g.mapUrl = "https://github.com/MorquinDevlar/GoMudUI/releases/latest/download/gomud.dat"
+	g.mapperUrl = "https://github.com/MorquinDevlar/GoMudUI/releases/latest/download/mudlet-mapper.mpackage"
+	g.mapperVersion = "1.0.0"
+	g.registeredUsers = make(map[int]bool)
+}
+
+// Public function that can be called from the GMCP module
+func (g *MudletClientModule) SendClientInfoExported(userId int) bool {
+	user := users.GetByUserId(userId)
+	if user == nil {
+		return false
+	}
+	
+	// Check if we've already sent info to this user
+	if _, ok := g.registeredUsers[userId]; ok {
+		return true // Already sent
+	}
+	
+	// Mark as registered
+	g.registeredUsers[userId] = true
+	
+	// Send initial client info
+	return g.sendClientInfo(user)
+}
+
+// RefreshConfig loads values from the module's config.yaml file
+func (g *MudletClientModule) RefreshConfig() {
```

Comment:
What happens if someone changing the config value while the mud is running, such as with the `server set` command? Would it get picked up?

We should probably have an event to handle this when it takes place, making it easier to do this sort of refresh...

You might consider just adding some accessor functions, so that it always fetches the current value:

```
func (g *MudletClientModule) GetUIUrl() string {
	if uiUrl, ok := g.plug.Config.Get(`Mudlet.UIUrl`).(string); ok  {
		return uiUrl
		mudlog.Info("MudletClientModule", "config", "Loaded UIUrl from config", "value", g.uiUrl)
	} 
	return ""
}
```